### PR TITLE
Package ppx_fast_pipe.0.0.1

### PR DESCRIPTION
**Pull request by IwanKaramazow**
### `ppx_fast_pipe.0.0.1`
Fast pipe, pipe first as a syntax transform
Provides a ppx to transform expressions containing the |. (Ocaml) or -> (Reason)
operator. Pipes the left side as first argument to the right side.

Example Reason:
--------------
```reason
/* validateAge(getAge(parseData(person))) */
person
->parseData
->getAge
->validateAge;

/* Some(preprocess(name)); */
name->preprocess->Some;

/* f(a, ~b, ~c) */
a->f(~b, ~c)
```

Example Ocaml:
--------------
```ocaml
(* validateAge (getAge (parseData person)) *)
person
|. parseData
|. getAge
|. validateAge

(* Some(preprocess name) *)
name |. preprocess |. Some;

(*f a ~b ~c *)
a |. f ~b ~c
```

---
* Homepage: https://github.com/IwanKaramazow/FastPipe
* Source repo: git://github.com/IwanKaramazow/FastPipe.git
* Bug tracker: https://github.com/IwanKaramazow/FastPipe

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
I'm sorry but I'm going to be picky. What's the goal of this package? What is the difference compared to the `|>` operator?

(also ppx packages tend to be named with `ppx_` at the beginning not the end)

**Comment by kit-ty-kate** [comment]
ah. It's the first argument.. mmh fine I guess

**Comment by kit-ty-kate** [comment]
Though regarding the name of the package it would be nice keeping the current "standard"

File: packages/fast_pipe_ppx/fast_pipe_ppx.0.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+maintainer: "Iwan Karamazow <m.falka@icloud.com>"
+authors: "Iwan Karamazow m.falka@icloud.com"
+license: "MIT"
+homepage: "https://github.com/IwanKaramazow/FastPipe"
+bug-reports: "https://github.com/IwanKaramazow/FastPipe"
+dev-repo: "git://github.com/IwanKaramazow/FastPipe.git"
+depends: [ 
+  "ocaml" {>= "4.02" & < "4.08"}
+  "reason" {>= "3.3.7"}
+  "dune"
```
Comment (kit-ty-kate) [inline packages/fast_pipe_ppx/fast_pipe_ppx.0.0.1/opam]: this dependency should be tagged as a `{build}` dependency

**Comment by IwanKaramazow** [comment]
Yes, the purpose is to pipe in the first argument. It also allows piping into variants which `|>` can't do.
There are also benefits regarding better inference for editor tooling. Will update the naming of the package. I followed `lwt_ppx` as an example and assumed that `_ppx` as suffix was the standard.

File: packages/fast_pipe_ppx/fast_pipe_ppx.0.0.1/opam (hunk 1 above)
Comment (IwanKaramazow) [reply packages/fast_pipe_ppx/fast_pipe_ppx.0.0.1/opam]: Fixed!

**Comment by IwanKaramazow** [comment]
@kit-ty-kate All items from the review have been addressed

**Comment by kit-ty-kate** [comment]
thanks!

**Pull request merged**